tracing-core/src/callsite.rs

```diff
@@ -235,7 +235,7 @@ mod inner {
 
 impl PartialEq for Identifier {
     fn eq(&self, other: &Identifier) -> bool {
-        self.0 as *const _ as *const () == other.0 as *const _ as *const ()
+        core::ptr::eq(self.0, other.0)
```

Comment:
the cast to `*const ()` is load-bearing here, since comparing trait object pointers is not a good idea. I think this needs to be 
```suggestion
        core::ptr::eq(self.0 as *const _ as *const (), other.0 as *const _ as *const ())
```